### DISCUSSION TITLE
About License

### DISCUSSION THREAD

**Discussion by kchan0514** [IfcOpenShell/IfcOpenShell#4102] [Q&A]
Hi team,

We are planning to release a plugin sorfware and this will be release via public Github repository.
The plugin uses Ifc OpenShell. In our plugin software,  users can download and install IfcOpenShell
And use the functions of it.

We will not include IfcOpenShell Source code. Just prepare download and install link to the binary
We will download and internally like below.
https://blenderbim.org/docs-python/ifcconvert/installation.html
After installing, our program execute commands to IfcOpenshell.

My question is, 
Is our SDK requred to be GPLv3 base?  As our original idea, we planned release our SDK as MIT License.

It would be very helpful if you could answer. Thank you.

**Comment by Moult** [comment]
I am not a lawyer :) It sounds like you're using the pre-built IfcConvert binary. Here's my understanding:

IfcConvert is LGPL v3-or-later licensed, not GPL v3. So in plain English:

 - You are free to have your code as the MIT license.
 - You need to attribute that IfcConvert is used in a place that users and devs can find it.
 - You need to include the LGPL v3 license as well in a place that users and devs can find it.
 - You can (optionally) [adopt a freckled duck](https://wetlands.org.au/living_wetlands/freckled-ducks-in-residence/) (Australia's rarest duck) in the Hunter Wetlands in Newcastle, Australia

For the lawyer text see https://www.gnu.org/licenses/lgpl-3.0.en.html#license-text

**Reply by aothms** [reply]
I agree. Also not a lawyer though.

The intention behind LGPL is twofold:

- allow end-users to inspect the source of the foss module and to relink and update it independently of your application. Both of these are guaranteed when using the prebuilt executables.

- If you were to make **modifications to IfcConvert / IfcOpenShell** you are required to provide these in some way (ideally a PR here on Github). This is the great thing about this license: that we all benefit from each others improvements, but can still build open or proprietary software around it.

**Reply by kchan0514** [reply]
I understand. Thank you so much!
